FLUID-6506: Migrated from Docpad to 11ty for GSoC 2020

sachin10101998 opened the pull request:
GSoC 2020 Project

BlueSlug commented on src/projects.html:
```diff
@@ -0,0 +1,136 @@
+                <dd>
+                    A collaboration of educational institutions focusing on audio and video content. Matterhorn is an Opencast project building an enterprise-level, open source podcast and rich media capture, processing and delivery system.
+                </dd>
+            </dt>
```
Should be `</dl>`

sachin10101998 replied:
Fixed

BlueSlug commented:
I also noticed that the spacing of elements differs between the live version running at https://fluidproject.org/ and the version I built locally. It would be good to know what's going on there, as the output should be identical (except for UIO which has been updated). It's subtle, but would be more apparent if you open these two sites and switch back and forth between them.

The live (DocPad) version:
![2020-05-28 fluidproject spacing difference - live site](https://user-images.githubusercontent.com/547933/83220108-1aedea80-a140-11ea-9b0b-03a9e0b645f2.jpg)

The dev (11ty) version:
![2020-05-28 fluidproject spacing difference - localhost](https://user-images.githubusercontent.com/547933/83220122-29d49d00-a140-11ea-914a-24c7a38b17b4.jpg)

BlueSlug commented on src/_includes/partials/header.html:
```diff
@@ -0,0 +1,44 @@
+        <ul class="right button-group">
+            
+                <li class="">
+                    <a href="{{ '/' | url }}index.html" class="button current">
```
The `current` item in the nav is hardcoded. This should change based on which file is being viewed.

![2020-05-28 fluidproject nav issue 001](https://user-images.githubusercontent.com/547933/83220237-77510a00-a140-11ea-80b9-479217ea5157.jpg)

sachin10101998 commented on README.md:
```diff
@@ -31,19 +24,6 @@ This is not an immediately deployable version of the website - [docpad](http://d
-
 ## License
 
 The Fluid Project website is available under [Creative Commons Attribution License](http://creativecommons.org/licenses/by/3.0/).
```
Fixed

sachin10101998 commented on src/projects.html:
```diff
@@ -0,0 +1,136 @@
+
+                <dt><a href="http://www.floeproject.org">Floe Project website<span class="fluid-web-external-link" role="presentation"></span></a></dt>
+                <dd>Supporting learners, educators and curriculum producers in achieving one-size-fits-one learning design for the full diversity of learners.</dd>
+            </ul>
```
Fixed

sachin10101998 commented on Gruntfile.js:
```diff
@@ -14,90 +16,120 @@ module.exports = function (grunt) {
-            all: ["package.json", ".eslintrc.json"]
+        lintAll: {
+            sources: {
+                json: ["package.json", ".eslintrc.json","./src/_data/*.json"],
```
Fixed

BlueSlug replied:
Please add Markdown linting back in and ensure all files are passing (you can do this work after deciding which news articles to keep and which to remove)

sachin10101998 commented on src/filters/w3-date-filter.js:
```diff
@@ -0,0 +1,8 @@
+/* eslint-env es6 */
+/* eslint-disable no-console */
```
Fixed

sachin10101998 commented on src/filters/markdown-filter.js:
```diff
@@ -0,0 +1,12 @@
+/* eslint-env es6 */
+/* eslint-disable no-console */
```
Fixed

sachin10101998 commented on src/filters/date-filter.js:
```diff
@@ -0,0 +1,17 @@
+/* eslint-env es6 */
+/* eslint-disable no-console */
```
Fixed

sachin10101998 commented on src/filters/date-filter.js:
```diff
@@ -0,0 +1,17 @@
+    const months = ["January", "February", "March", "April", "May", "June", "July", "August", "September", "October", "November", "December"];
+    const dayWithSuffix = appendSuffix(dateObject.getDate());
+
+    return `${dayWithSuffix} ${months[dateObject.getMonth()]} ${dateObject.getFullYear()}`;
```
It was caught. But I am already using double-quotes.

sachin10101998 commented on src/filters/w3-date-filter.js:
```diff
@@ -0,0 +1,8 @@
+/* eslint-env es6 */
```
Fixed

sachin10101998 commented on src/filters/date-filter.js:
```diff
@@ -0,0 +1,17 @@
+/* eslint-env es6 */
```
Fixed

sachin10101998 commented on src/filters/markdown-filter.js:
```diff
@@ -0,0 +1,12 @@
+/* eslint-env es6 */
```
Fixed

sachin10101998 commented on src/assets/js/floe.js:
```diff
@@ -0,0 +1,31 @@
+// Declare dependencies
+/*global fluid, jQuery*/
+"use strict";
+var floe = floe || {};
```
Fixed

sachin10101998 commented on src/assets/js/floe.js:
```diff
@@ -0,0 +1,31 @@
+/*
```
Fixed

sachin10101998 commented on src/assets/js/floe.js:
```diff
@@ -0,0 +1,31 @@
+Copyright 2017 OCAD University
+Copyright 2020 OCAD University
+
+Licensed under the Educational Community License (ECL), Version 2.0 or the New
```
Fixed

sachin10101998 commented on src/assets/js/floe.js:
```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2014 OCAD University
```
Fixed

sachin10101998 commented on src/_includes/partials/header.html:
```diff
@@ -0,0 +1,44 @@
+        </div>
+        <h1><a href="/"><span class="fluid-web-logo-star" role="presentation"></span>fluid</a></h1>
+    </header>
+    <nav  id="fluid-web-navigation" class="small-12 columns fluid-web-nav">
```
Fixed

sachin10101998 commented on src/_includes/partials/header.html:
```diff
@@ -0,0 +1,44 @@
+<div class="row">
```
Fixed

sachin10101998 commented on src/_includes/layouts/post.html:
```diff
@@ -0,0 +1,30 @@
+                {{ content }}
+                {% include partials/footer.html %}
+            </div>
+        </body>
```
Fixed

sachin10101998 commented on src/_includes/layouts/post.html:
```diff
@@ -0,0 +1,30 @@
+        <script type="text/javascript" src="{{ '/' | url }}lib/infusion/infusion-all.js"></script>
+        <script type="text/javascript" src="{{ '/' | url }}assets/js/floe.js"></script>
+        {% include partials/foundationflexgrid.html %}
+        </head>
```
Fixed

sachin10101998 commented on src/_includes/layouts/default.html:
```diff
@@ -0,0 +1,36 @@
+        {% include partials/uiofloe.html %}
+        <div class="fluid-web">
+            
+                {% include partials/header.html %}
```
Fixed

sachin10101998 commented on src/_includes/layouts/default.html:
```diff
@@ -0,0 +1,36 @@
+            </div>
+        
+    </body>   
+    </head>
```
Fixed

sachin10101998 commented on src/_includes/partials/fluidcommunityontheweb.html:
```diff
@@ -1,3 +1,4 @@
+
```
Fixed

sachin10101998 commented on src/_includes/partials/fluidcommunityontheweb.html:
```diff
@@ -11,4 +12,4 @@ <h2 class="fluid-web-deemphasized-text">The Fluid Community on the Web</h2>
         </ul>
     </div>
-</div>
+    
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -2,25 +2,18 @@
 1. Get the required node modules: `npm install`
-2. Run docpad from the fluid-website directory `npm run docpad`.
+2. Run eleventy from the fluid-website directory `npm run eleventyport`.
 3. Open `http://localhost:9778/` to see the website. 
```
Fixed

sachin10101998 commented on src/_data/README.md:
```diff
@@ -0,0 +1,2 @@
+**_data***
+For information about this directory, see: (Global Data Files)[https://www.11ty.dev/docs/data-global/]
```
Fixed

sachin10101998 commented on src/_data/README.md:
```diff
@@ -0,0 +1,2 @@
+**_data***
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -31,19 +24,6 @@ This is not an immediately deployable version of the website - [docpad](http://d
   will not appear when deployed locally or when deployed through a gh-pages (sub) project. To test the 404 error page,
```
Fixed

sachin10101998 commented on AUTHORS.md:
```diff
@@ -0,0 +1,17 @@
+This is the list of FLOE Project copyright holders. It does not list all individual contributors because some have assigned
+copyright to an institution, only made minor changes, or their contributions no longer appear in the codebase.
+Please see the version control system's revision history for details on contributions.
+
```
Fixed. Am not sure of what other authors should be listed here. Maybe @colinbdclark can help

BlueSlug replied:
If the answer is "there are no other authors", then that's fine too :)

sachin10101998 replied:
Fixed then.

sachin10101998 commented on package.json:
```diff
@@ -1,35 +1,42 @@
+    },
+    "homepage": "https://github.com/fluid-project/fluidproject.org#readme",
+    "dependencies": {
+        "@11ty/eleventy": "^0.11.0",
```
@jobara thoughts?

sachin10101998 replied:
Fixed

sachin10101998 commented on src/404.html:
```diff
@@ -0,0 +1,7 @@
+permalink: 404.html
+---
+<h1>{{ title }}</h1>
+<p>The page you were looking for does not exist yet. Go back to <a href="{{ '/' | url }}">home page?</a></p>
```
Fixed

sachin10101998 commented on src/posts/2007-05-04-Better_Web_Experiences.md:
```diff
@@ -0,0 +1,32 @@
+date: '2007-05-04'
+tags: post
+---
+<section class="row">
```
Fixed

sachin10101998 commented on .gitignore:
```diff
@@ -1,5 +1,2 @@
-node_modules/
-src/files/lib
-.project
-package-lock.json
```
Most of the files would be required if someone tries to build the site locally. I'm not sure about package-lock.json though. @jhung thoughts?

jhung replied:
I don't have a strong opinion, but we have traditionally ignored `package-lock.json`.

sachin10101998 replied:
Fixed. Pinned the dependency in package.json file and deleted package-lock.json

jobara replied:
Just to add some of my thoughts and feedback about package.json for historical purposes. If you don't pin the versions, in general you'll need to use `npm ci` instead of `npm install` when installing dependencies. 

From what I understand the contrast is that pinning to specific versions you always get that versions. If you rely on the package-lock you'll get the latest considered release (based on how you've specified the version in the package.json file). This means that two people using the same code base may get different versions. In general that's likely going to be patch release versions. That's a benefit on one hand because it should just be bug fixes, but on the other hand any variation in codebase can make testing, debugging, and communicating issues more complicated.

jhung commented:
There's a bug where loading the root of the site (i.e. `localhost:9778/`) incorrectly sets News as the current navigation item, but should be Home.

![image](https://user-images.githubusercontent.com/564762/83293277-8e800e00-a1b9-11ea-85fc-3c8865df45aa.png)

Loading the site using `localhost:9778/index.html` does not cause this problem.

jhung commented:
The Infusion directory `src/lib/infusion` is being tracked, but I wonder if this is necessary or desired since the contents of that directory is copied from `./node_modules/infusion/dist/`. @BlueSlug thoughts?

sachin10101998 commented:
@jhung I fixed that bug where the default page showed news as highlighted or selected page. I've also removed the package lock json file and pinned dependencies in package.json file. Can we merge it now?

gtirloni commented on README.md:
```diff
@@ -1,49 +1,26 @@
-# To deploy to a personal webserver
+## To deploy to a personal webserver
 
 1. Run: `npm run generate`
```
```
$ npm run generate
npm ERR! missing script: generate
```

sachin10101998 replied:
Fixed

BlueSlug commented:
> The Infusion directory `src/lib/infusion` is being tracked, but I wonder if this is necessary or desired since the contents of that directory is copied from `./node_modules/infusion/dist/`. @BlueSlug thoughts?

If Infusion is being pulled in via npm, then I agree that it isn't necessary to track this directory unless it's required for the build process.

gtirloni commented on package.json:
```diff
@@ -1,35 +1,41 @@
+        "type": "git",
+        "url": "git+https://github.com/fluid-project/fluidproject.org.git"
+    },
+    "author": "OCAD Univesity",
```
University

sachin10101998 replied:
fixed

BlueSlug commented on .eleventy.js:
```diff
@@ -0,0 +1,85 @@
+      ...collection.getFilteredByGlob('./src/posts/*.md').filter(livePosts)
+    ];
+  });
+  // The following xollection is ues to distribute posts into different pages. However, the default pagination has not been set in floeproject.org and all posts are shown on single page
```
"xollection" should be "collection"
"floeproject.org" should be "fluidproject.org"

sachin10101998 replied:
Fixed

BlueSlug commented on AUTHORS.md:
```diff
@@ -0,0 +1,19 @@
+## Contributors
+
+Individual contributions can be viewed on the
+[Contributors](https://github.com/fluid-project/floeproject.org/graphs/contributors) page, or through the version control
```
This should be the Fluid Project site repo, not FLOE Project

sachin10101998 replied:
Fixed

BlueSlug commented on src/_includes/partials/header.html:
```diff
@@ -0,0 +1,42 @@
+    </header>
+    <nav id="fluid-web-navigation" class="small-12 columns fluid-web-nav">
+        <ul class="right button-group" id="fluid-web-nav">
+            <li class="">
```
These empty `class` attributes can be removed.

Also, these list items probably don't need to be spread out across so many lines. They can be condensed to 5 lines and still remain readable.

sachin10101998 replied:
Fixed

BlueSlug commented on src/_includes/partials/header.html:
```diff
@@ -0,0 +1,42 @@
+            </li>
+            
+            <li class="">
+                <a href="{{ '/' | url }}about.html" class="button " id="aboutnav">
```
Please remove the spaces after `button` in `class="button "` (more instances of this issue below)

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -0,0 +1,34 @@
+      <ol>
+         <li> <a href="http://issues.fluidproject.org/browse/FLUID-3077">Text inside inline editor is being lost after canceling the text edit.</a></li>
+         <li>
+            <a style="" href="/browse/FLUID-1320">Typing &quot;Click here to edit&quot; is equivalent to erasing all the text</a>
```
Please remove the `style` attribute

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -0,0 +1,34 @@
+         <li>
+            <a style="" href="/browse/FLUID-1320">Typing &quot;Click here to edit&quot; is equivalent to erasing all the text</a>
+         </li>
+         <li> <a   href="/browse/FLUID-2967">Item will drop to the right of an element when it should have dropped at the left.</a></li>
```
please remove the extra spaces after `<a`

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -0,0 +1,34 @@
+            <a style="" href="/browse/FLUID-1320">Typing &quot;Click here to edit&quot; is equivalent to erasing all the text</a>
+         </li>
+         <li> <a   href="/browse/FLUID-2967">Item will drop to the right of an element when it should have dropped at the left.</a></li>
+         <li><a style="" href="/browse/FLUID-3054">tiny_mce Inline Edit text editor fails when used with advanced theme</a></li>
```
Please remove the empty `style` attribute

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+The Fluid community consists of an international team of partners, individuals, and institutions focused on designing inclusive, ﬂexible, customizable, user-centered interfaces. 
+
+
+The present version of Fluid Project uses 11ty.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+
+The present version of Fluid Project uses 11ty.
+
+**To run the project locally**:
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+# `Fluid Project`
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+The present version of Fluid Project uses 11ty.
+
+**To run the project locally**:
+1. Clone the project locally.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+
+**To run the project locally**:
+1. Clone the project locally.
+2. Go to the folder directory where you cloned the project in terminal.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+**To run the project locally**:
+1. Clone the project locally.
+2. Go to the folder directory where you cloned the project in terminal.
+3. Get the required node modules: `npm install`
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+5. Open `http://localhost:9778/` to see the website. 
+
+**The directory structure is as follows:**
+* `_.github` - It includes the issue and PR templates for the FLOE Project Repository.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+
+**The directory structure is as follows:**
+* `_.github` - It includes the issue and PR templates for the FLOE Project Repository.
+* `src` - The src folder contains all the files needed to build the site using 11ty. It contains content of all the blog posts/news articles, CSS files, Index Pages, layouts for pages, partial templates, Javascript codes etc. 
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+* `_.github` - It includes the issue and PR templates for the FLOE Project Repository.
+* `src` - The src folder contains all the files needed to build the site using 11ty. It contains content of all the blog posts/news articles, CSS files, Index Pages, layouts for pages, partial templates, Javascript codes etc. 
+* `.eleventy.js` - The Configuration file used by 11ty in order to build the site. Specifies the collections for 11ty, passthroughcopy files, and other settings including the source folder for input and output.
+* `.eleventyignore` - Contains folder which are ignored by Eleventy while building the site.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+**The directory structure is as follows:**
+* `_.github` - It includes the issue and PR templates for the FLOE Project Repository.
+* `src` - The src folder contains all the files needed to build the site using 11ty. It contains content of all the blog posts/news articles, CSS files, Index Pages, layouts for pages, partial templates, Javascript codes etc. 
+* `.eleventy.js` - The Configuration file used by 11ty in order to build the site. Specifies the collections for 11ty, passthroughcopy files, and other settings including the source folder for input and output.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+4. Run eleventy from the fluid-website directory `npm run eleventyport`.
+5. Open `http://localhost:9778/` to see the website. 
+
+**The directory structure is as follows:**
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+* `src` - The src folder contains all the files needed to build the site using 11ty. It contains content of all the blog posts/news articles, CSS files, Index Pages, layouts for pages, partial templates, Javascript codes etc. 
+* `.eleventy.js` - The Configuration file used by 11ty in order to build the site. Specifies the collections for 11ty, passthroughcopy files, and other settings including the source folder for input and output.
+* `.eleventyignore` - Contains folder which are ignored by Eleventy while building the site.
+* `.gitignore` - List of folders not to be uploaded to Github. contains node_modules and dist folder, which is the folder where 11ty saves output files.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+* `.eleventy.js` - The Configuration file used by 11ty in order to build the site. Specifies the collections for 11ty, passthroughcopy files, and other settings including the source folder for input and output.
+* `.eleventyignore` - Contains folder which are ignored by Eleventy while building the site.
+* `.gitignore` - List of folders not to be uploaded to Github. contains node_modules and dist folder, which is the folder where 11ty saves output files.
+* `.eslintrc.json` - EsLint Configuration data.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+* `.eleventyignore` - Contains folder which are ignored by Eleventy while building the site.
+* `.gitignore` - List of folders not to be uploaded to Github. contains node_modules and dist folder, which is the folder where 11ty saves output files.
+* `.eslintrc.json` - EsLint Configuration data.
+* `.eslintignore` - EsLint exempted files.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+* `.gitignore` - List of folders not to be uploaded to Github. contains node_modules and dist folder, which is the folder where 11ty saves output files.
+* `.eslintrc.json` - EsLint Configuration data.
+* `.eslintignore` - EsLint exempted files.
+* `.nojekyll` - Required to let Github know that the site does not use Jekyll.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+* `.eslintrc.json` - EsLint Configuration data.
+* `.eslintignore` - EsLint exempted files.
+* `.nojekyll` - Required to let Github know that the site does not use Jekyll.
+* `AUTHORS.md` - List of copyright holders and Contributors
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+* `.nojekyll` - Required to let Github know that the site does not use Jekyll.
+* `AUTHORS.md` - List of copyright holders and Contributors
+* `Gruntfile.js` - Grunt file used to copy data from node_modules and do linting process.
+* `package.json` - Pinned list of dependencies required to build the site.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+    - Improving the User interface of submitted answers.
+    - Search functionality among the site data.
+    - Support for Comments.
+2. *Submitting new blog posts or news articles*
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+    - Search functionality among the site data.
+    - Support for Comments.
+2. *Submitting new blog posts or news articles*
+ - Answer submission through Github:
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+
+
+**Contributions to the project can be made in following ways:**
+1. *Contribution towards building features for the project.*
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+2. *Submitting new blog posts or news articles*
+ - Answer submission through Github:
+   - Go to the `src/posts` folder
+   - Create a new Markdown file with title in the following format - `YYYY-MM-DD-titleOfNewsArticle.md`.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+   - Create a new Markdown file with title in the following format - `YYYY-MM-DD-titleOfNewsArticle.md`.
+   - Use the YYYY-MM-DD-newsarticlename.md file as template. 
+   - Update the date, title and permalink in the front matter.
+   - Your article must have a specific template as mentioned below to get merged. make sure to follow the template while submitting the answer.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+      `---`
+      `Content Data`
+
+If you still have any qeries/doubts regarding the project, feel free to join our IRC Channel #fluid-work on Freenode.
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,62 @@
+
+
+
+
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -1,49 +1,26 @@
 # Fluid Project Website
 
-This repository contains the files needed to build a copy of the Fluid Project website.
+1. This repository contains the files needed to build a copy of the Fluid Project website.
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -1,49 +1,26 @@
-   may deploy to the wrong location.
-2. Deploy to gh-pages, run: `npm run deploy`. By doing this, docpad will generate the site to the   
-   remote gh-pages branch.
+2. Run [eleventy](http://11ty.dev) from the fluid-website directory `npm run eleventyport`.
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -1,49 +1,26 @@
-* eco plugin - https://github.com/docpad/docpad-plugin-eco
-* gh-pages plugin - https://github.com/docpad/docpad-plugin-ghpages
-* redirector plugin - https://www.npmjs.org/package/docpad-plugin-redirector
+1. Modifications can be done to any source file or directory except for the contents of the dist/ directory.
```
Fixed

sachin10101998 commented on README.md:
```diff
@@ -1,49 +1,26 @@
-* gh-pages plugin - https://github.com/docpad/docpad-plugin-ghpages
-* redirector plugin - https://www.npmjs.org/package/docpad-plugin-redirector
+1. Modifications can be done to any source file or directory except for the contents of the dist/ directory.
+2. The dist directory are not to be versioned since it is the generated output made by 11ty from the source files.
```
Fixed

sachin10101998 commented on Gruntfile.js:
```diff
@@ -1,8 +1,10 @@
 /*
-Copyright 2017 OCAD University
-
+Copyright 2011-2020 OCAD University
```
Fixed

sachin10101998 commented on Gruntfile.js:
```diff
@@ -14,90 +16,121 @@ module.exports = function (grunt) {
+                md: ["./*.md","./src/posts/*.md"]
+            }
         },
         clean: {
```
Fixed

sachin10101998 commented on src/404.html:
```diff
@@ -0,0 +1,12 @@
+---
+layout: layouts/default
+title: Oops! Not Found
```
Fixed

sachin10101998 commented on src/_includes/layouts/post.html:
```diff
@@ -0,0 +1,31 @@
+        <meta content="text/html; charset=utf-8" http-equiv="Content-Type" />
+        <meta content="width=device-width, initial-scale=1.0" name="viewport">
+        <link type="image/x-icon" href="{{ '/' | url }}assets/images/favicon.ico" rel="shortcut icon"/>
+        <!-- Foundation flex grid -->
```
Fixed

sachin10101998 commented on src/_includes/layouts/post.html:
```diff
@@ -0,0 +1,31 @@
+                uio.setupUIO();
+            });
+        </script>
+            {% include partials/uio.html %}
```
Fixed

sachin10101998 commented on src/assets/js/nav.js:
```diff
@@ -0,0 +1,28 @@
+var linkk=window.location.pathname;
```
Fixed

sachin10101998 commented on src/assets/js/nav.js:
```diff
@@ -0,0 +1,28 @@
+var infusionnav=document.getElementById("infusionnav");
+var newsnav=document.getElementById("newsnav");
+if(linkk=="/"|| linkk=="/index.html")
+{
```
Fixed

sachin10101998 commented on src/assets/js/uio.js:
```diff
@@ -0,0 +1,24 @@
+/*
+Licensed under the New BSD license. You may not use this file except in 
+compliance with the license.
+
```
Fixed

sachin10101998 commented on src/assets/js/uio.js:
```diff
@@ -0,0 +1,24 @@
+
+*/
+
+// Declare dependencies
```
Fixed

sachin10101998 commented on src/assets/js/uio.js:
```diff
@@ -0,0 +1,24 @@
+*/
+
+// Declare dependencies
+/*global fluid, jQuery*/
```
Fixed

sachin10101998 commented on src/assets/stylesheets/foundation/README.md:
```diff
@@ -0,0 +1,10 @@
+  * Grid max-width was set to 76rem
+* All files removed from the extracted ZIP file except *foundation.css*
+* The Normalize.css include in *foundation.css* by default was stripped from the file due to referencing the version of Normalize include with Infusion.
+* Foundation.css had been edited in order to suit needs of FLOEproject.org
```
Fixed

sachin10101998 commented on src/assets/stylesheets/foundation/README.md:
```diff
@@ -0,0 +1,10 @@
+* Build was downloaded from http://foundation.zurb.com/sites/download.html/#customizeFoundation with all components unchecked except for "Flex Grid"
+  * Grid max-width was set to 76rem
+* All files removed from the extracted ZIP file except *foundation.css*
+* The Normalize.css include in *foundation.css* by default was stripped from the file due to referencing the version of Normalize include with Infusion.
```
Fixed

sachin10101998 commented on src/posts/YYYY-MM-DD-newsarticletemplate.md:
```diff
@@ -0,0 +1,8 @@
+tags: post
+permalink: YYYY-MM-DD-newsArticleName.html
+---
+//Content Data
```
Fixed

sachin10101998 commented on src/posts/YYYY-MM-DD-newsarticletemplate.md:
```diff
@@ -0,0 +1,8 @@
+title: Title of News Article
+date: '2020-01-01'
+tags: post
+permalink: YYYY-MM-DD-newsArticleName.html
```
Sure, fixed.

sachin10101998 commented on src/projects.html:
```diff
@@ -0,0 +1,134 @@
+            <h3>Video Player</h3>
+            <p>An accessible, transformable, and open source HTML5 video player.</p>
+            <a href="http://build.fluidproject.org/videoPlayer/demos/Mammals.html" class="row"><span class="fluid-web-showcase-item-text">Video Player Demo<span class="fluid-web-external-link" role="presentation"></span></span></a>
+
```
Fixed

sachin10101998 commented on src/projects.html:
```diff
@@ -0,0 +1,134 @@
+            <p>
+                The Fluid Project is designing the architecture for the GPII and building the core infrastructure. Fluid is involved in a number of projects that will harness the GPII to empower users:
+            </p>
+
```
Fixed

sachin10101998 commented on src/filters/w3-date-filter.js:
```diff
@@ -0,0 +1,7 @@
+/* eslint-disable */
```
Fixed

BlueSlug replied:
Please remove this line

BlueSlug replied:
This line is still present

sachin10101998 commented on src/filters/markdown-filter.js:
```diff
@@ -0,0 +1,11 @@
+/* eslint-disable */
```
Fixed

BlueSlug replied:
Please remove this line

sachin10101998 commented on src/filters/date-filter.js:
```diff
@@ -0,0 +1,16 @@
+/* eslint-disable */
```
Fixed

BlueSlug replied:
Please remove this line

sachin10101998 commented on src/projects.html:
```diff
@@ -0,0 +1,134 @@
+                    <p class="fluid-web-projects-partners">
+                        In partnership with Detroit Institute of Arts - McCord Museum of Canadian History - Museum of the Moving Image - Open University of Catalonia - Simon Fraser University - Cambridge University - University of Colorado - University of Toronto
+                    </p>
+
```
Fixed

BlueSlug replied:
This line is still present, please remove it. Please also be careful of writing "fixed" on something that has not been fixed.

sachin10101998 commented on src/assets/stylesheets/foundation/README.md:
```diff
@@ -0,0 +1,10 @@
+*Components Included*: Flex Grid
+
+* Build was downloaded from http://foundation.zurb.com/sites/download.html/#customizeFoundation with all components unchecked except for "Flex Grid"
+  * Grid max-width was set to 76rem
```
Fixed

BlueSlug replied:
Please explain why the grid `max-width` was set to `76rem`

sachin10101998 commented on Gruntfile.js:
```diff
@@ -14,90 +16,121 @@ module.exports = function (grunt) {
         },
         copy: {
             // Copy external front end dependencies into appropriate directories
             frontEndDependencies: {
```
It wouldn't be a good practice because the dist directory would be overwritten in case someone does npm run build in order to get the files from the `dist` folder to upload to some personal server.

sachin10101998 commented:
@BlueSlug 
I have addressed all the review comments and made the changes as required. 
1) nojekyll file has been removed
2) Added code to analytics.js as prescribed by Gtirloni
3) Made all conversions from floe to fluid.
4) Added the right copyright template to all original JS files.
5) default.html and post.html are almost identical but when I try to merge them, infusion uio breaks for some reason unknown and it doesn't work and console shows errors. I have tried all methods to combine them but until and unless they are in different formats with different names, infusion breaks. Hence I kept them separate.
6) The inconsistency in the live site and migrated site is due to the fact that I changed the UIO preferences in order to keep it in sync with floe project. It made no sense to have different stylings for the same thing and a consistent style for the uio button provided with consistency. Sue to different shapes of the button, the site shifts a bit low compared to the original site, but the migrated site is in accordance with floe project.
7) Markdown inconsistencies have been caused due to usage of HTML tags in the MD files. These HTML tags come from the original news articles. i really don't know how to go around this. We can either remove the posts folder from markdown linting or we'll have to rewrite those posts in accordance with markdown supported tags.

BlueSlug commented on src/assets/js/nav.js:
```diff
@@ -0,0 +1,31 @@
+/*
```
I got 28 linter errors for this file. Please make sure the linter is passing without errors before committing.

The issues in this file can mostly be fixed with these two suggestions:
- there needs to be space between the variable names and assigned values and the `=` operator
  - e.g. `var link=window.location.pathname;` should be `var link = window.location.pathname;`
- try not to use `==` for comparison in JavaScript, as it may not always produce the result you intend it to. Please use `===` instead.
  - Here's some more info on this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

sachin10101998 replied:
Which linter are you using? The file is added in the sources list for linting and the linter shows no error.

sachin10101998 replied:
I've made the changes you pointed out in particular though.

sachin10101998 replied:
The problem was I forgot the push the changes I made on my local repo. Hence, some of the previous review changes were not updated. I have pushed all changes now.

BlueSlug commented on src/assets/js/analytics.js:
```diff
@@ -0,0 +1,19 @@
+/*
```
I got 30 linter errors for this file. Please make sure the linter is passing without errors before committing. Readability is very important for ease of maintenance.

Some quick fixes:
- make sure all strings use double-quotes `"` instead of single quotes `'`
- ensure the indentation matches what `eslint-config-fluid` specifies
- add spaces before and after operators like `=`, `||`, `&&`

sachin10101998 replied:
fixed

BlueSlug commented on src/filters/date-filter.js:
```diff
@@ -0,0 +1,24 @@
+*/
+/* eslint-disable */
+"use strict";
+// this function is strict...
```
Please remove this comment

sachin10101998 replied:
fixed

BlueSlug commented on .eleventy.js:
```diff
@@ -0,0 +1,85 @@
+      ...collection.getFilteredByGlob('./src/posts/*.md').filter(livePosts)
+    ];
+  });
+  // The following collection is ues to distribute posts into different pages. However, the default pagination has not been set in fluidproject.org and all posts are shown on single page
```
Please change "ues" to "used". Please also move the second sentence to a new comment below this one

sachin10101998 replied:
fixed

BlueSlug commented on .eleventy.js:
```diff
@@ -0,0 +1,85 @@
+          // Provides the 404 content without redirect.
+          res.write(content_404);
+          // Add 404 http status code in request header.
+          // res.writeHead(404, { "Content-Type": "text/html" });
```
Do we intend to use this commented out line at any point in the future?

sachin10101998 replied:
No. Deleted the comment.

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+1. Run eleventy from the fluid-website directory `npm run eleventyport`.
+1. Open `http://localhost:9778/` to see the website. 
+
+**Contributions to the project can be made in following ways:**
```
Fixed.

sachin10101998 commented on .eleventyignore:
```diff
@@ -0,0 +1,3 @@
+README.md
+src/lib/
+src/assets/stylesheets/foundation/README.me
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+The Fluid Project website is built using 11ty
+
+## To run the project locally:
+1. Clone the project locally using `git clone https://github.com/fluid-project/fluidproject.org.git`
```
Not a problem. Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+  - Add Title and Description as per given in the template.
+  - Click on 'Submit New Issue'.
+
+**The directory structure is as follows:**
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+| `_.github`        | A list of files and folders which are ignored. It includes the issue and Pull Requests templates for the fluid Project Repository along with node_modules.                                                                          |
+| `src`             | Contains all the files needed to build the site using 11ty. This includes the content for all pages and news articles, CSS files, layout and partial templates, and JavaScript files.                                               |
+| `.eleventy.js`    | The Configuration file used by 11ty in order to build the site. Specifies the collections for 11ty, [passthrough copy files](https://www.11ty.dev/docs/copy/), and other settings including the source folder for input and output. |
+| `.eleventyignore` | Contains folder which are ignored by Eleventy while building the site.                                                                                                                                                              |
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+
+**The directory structure is as follows:**
+
+| Folder            | Description                                                                                                                                                                                                                         |
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+| `.eleventy.js`    | The Configuration file used by 11ty in order to build the site. Specifies the collections for 11ty, [passthrough copy files](https://www.11ty.dev/docs/copy/), and other settings including the source folder for input and output. |
+| `.eleventyignore` | Contains folder which are ignored by Eleventy while building the site.                                                                                                                                                              |
+| `.gitignore`      | A list of files and folders which won't be tracked by Git.                                                                                                                                                                          |
+| `.eslintrc.json`  | [EsLint configuration data](https://eslint.org/docs/user-guide/configuring).                                                                                                                                                        |
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+| `.eslintignore`   | A list of files and folders which won't be passed through ESLint                                                                                                                                                                    |
+| `AUTHORS.md`      | List of copyright holders and contributors                                                                                                                                                                                          |
+| `Gruntfile.js`    | Grunt file used to copy data from node_modules and do linting process.                                                                                                                                                              |
+| `package.json`    | Pinned list of dependencies required to build the site.                                                                                                                                                                             |
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+    - Improving the User interface of submitted answers.
+    - Search functionality among the site data.
+    - Support for Comments.
+### Submitting new blog posts or news articles*
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+
+**Contributions to the project can be made in following ways:**
+
+### Contribution towards building features for the project.*
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+
+| Folder            | Description                                                                                                                                                                                                                         |
+|-------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `_.github`        | A list of files and folders which are ignored. It includes the issue and Pull Requests templates for the fluid Project Repository along with node_modules.                                                                          |
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+   - Use the YYYY-MM-DD-newsarticlename.md file as template. 
+   - Update the date, title and permalink in the front matter.
+   - Your article must use the following template for 11ty to process it properly.
+   - The template is as follows:
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+   - Create a new Markdown file with title in the following format - `YYYY-MM-DD-{News Article Name with underscores in place for spaces}.md`.
+   - Use the YYYY-MM-DD-newsarticlename.md file as template. 
+   - Update the date, title and permalink in the front matter.
+   - Your article must use the following template for 11ty to process it properly.
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+[#fluid-work](https://wiki.fluidproject.org/display/fluid/IRC+Channel) IRC channel,
+[fluid-work](https://lists.idrc.ocad.ca/mailman/listinfo/fluid-work) mailing list, and/or on the PR itself.
+
+If you still have any qeries/doubts regarding the project, feel free to visit [Get Involved](https://wiki.fluidproject.org/display/fluid/Get+Involved) page of our wiki to connect with us.
```
fixed

sachin10101998 commented on README.md:
```diff
@@ -23,40 +22,17 @@ The website will be available at [http://localhost:8000](http://localhost:8000)
-* eco plugin - https://github.com/docpad/docpad-plugin-eco
-* gh-pages plugin - https://github.com/docpad/docpad-plugin-ghpages
-* redirector plugin - https://www.npmjs.org/package/docpad-plugin-redirector
+-  Modifications can be done to any source file or directory except for the contents of the `dist/` directory.
```
fixed

sachin10101998 commented on src/_includes/partials/header.html:
```diff
@@ -0,0 +1,42 @@
+                <a href="{{ '/' | url }}about.html" class="button" id="aboutnav">
+                    About
+                </a>
+                </li>
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,70 @@
+**Contributions to the project can be made in following ways:**
+
+### Contribution towards building features for the project.*
+  - There are several features that still require attention.
```
fixed

sachin10101998 commented:
@BlueSlug I've made the required changes. Last time, I forgot to push the changes after making them on my local and marked them as fixed in the review comments above. I have pushed all the changes this time. I am sorry for that.

BlueSlug commented on AUTHORS.md:
```diff
@@ -0,0 +1,19 @@
+[Contributors](https://github.com/fluid-project/fluidproject.org/graphs/contributors) page, or through the version control
+system's revision history.
+
+**Note**: Individual authors may not hold copyright. See above "Copyright Holders" section for more information._
```
Please remove the underscore at the end of this line

sachin10101998 replied:
fixed

BlueSlug commented on src/filters/markdown-filter.js:
```diff
@@ -0,0 +1,19 @@
+
```
Please remove this blank line at the top

sachin10101998 replied:
fixed

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/fluid-project/fluidproject.org?pullRequest=56) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/fluid-project/fluidproject.org?pullRequest=56) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: sachin10101998<br/>:white_check_mark: gtirloni<br/>:x: Sachin Chopra<br/><hr/>**Sachin Chopra** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/fluid-project/fluidproject.org?pullRequest=56) it.</sub>

BlueSlug commented on src/posts/2008-12-19-Fluid_Infusion_v0_6_released.md:
```diff
@@ -0,0 +1,37 @@
+
+### What's New in This Release
+
+New features of this release include:\
```
Please remove the extra `\` at the end of this line

sachin10101998 replied:
Fixed.

BlueSlug commented on src/posts/2009-05-27-Fluid_Infusion_1_0_released.md:
```diff
@@ -0,0 +1,60 @@
+designers, Infusion components are built from the ground up with
+accessibility in mind. All of our designs can be used with assistive
+technologies, are fully controllable with the keyboard, and can be
+transformed to suit your users’ personal needs. \
```
Please remove the extra `\` at the end of this line

sachin10101998 replied:
fixed

BlueSlug commented on src/posts/2007-08-22-Off_to_a_Running_Start_with_UX_Walkthroughs.md:
```diff
@@ -0,0 +1,40 @@
+    area](http://wiki.fluidproject.org/display/fluid/User+Experience+Walkthroughs)
+    of our wiki as it is in need of organization and consolidation. If
+    you’ve got time for some quick editing and re-organization, please
+    feel free! \
```
Please remove the extra `\` at the end of this line

sachin10101998 replied:
fixed

BlueSlug commented on src/posts/2009-06-02-Fluid_Infusion_1_1_released.md:
```diff
@@ -0,0 +1,61 @@
+designers, Infusion components are built from the ground up with
+accessibility in mind. All of our designs can be used with assistive
+technologies, are fully controllable with the keyboard, and can be
+transformed to suit your users’ personal needs. \
```
Please remove the extra `\` at the end of this line

sachin10101998 replied:
fixed

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -0,0 +1,57 @@
+    text](/browse/FLUID-1320)
+3. [Item will drop to the right of an element when it should have
+    dropped at the left.](/browse/FLUID-2967)
+4. [tiny\_mce Inline Edit text editor fails when used with advanced
```
Please remove the `\` on this line

sachin10101998 replied:
fixed

BlueSlug commented on src/filters/w3-date-filter.js:
```diff
@@ -0,0 +1,15 @@
+See the AUTHORS.md file at the top-level directory of this distribution and at
+https://github.com/fluid-project/fluidproject.org/master/AUTHORS.md.
+*/
+/* eslint-disable */
```
Please remove this line...

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2007-05-04-Better_Web_Experiences.md:
```diff
@@ -0,0 +1,94 @@
+date: '2007-05-04'
+tags: post
+---
+The Fluid Project Receives \$2.5 Million for Interface Design: Better Web Experience for Users of Open Source Software {.fluid-web-emphasized-text}
```
Is this `\` necessary? there are two others in this file, too, on lines 16 and 90.

sachin10101998 replied:
fixed

sachin10101998 commented:
@BlueSlug , I'll be redoing the posts markdown files again as the current structure totally disturbs the styling. I'll make sure the linter passes.

BlueSlug commented on src/posts/2007-12-01-Fluid 0_1_is_Here.md:
```diff
@@ -5,41 +5,38 @@ permalink: 2007-12-01-Fluid 0_1_is_Here.html
+   <div class="medium-6 columns">
+      <p>The Fluid Project is pleased to announce the first release of the Fluid  component
+       library and User Experience (UX) Toolkit.</p>
+      <h3>What&#39;s New in This Release?</strong></h3>
```
Please remove `</strong>` from this line

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2007-12-01-Fluid 0_1_is_Here.md:
```diff
@@ -5,41 +5,38 @@ permalink: 2007-12-01-Fluid 0_1_is_Here.html
+       library and User Experience (UX) Toolkit.</p>
+      <h3>What&#39;s New in This Release?</strong></h3>
+      <ul>
+         <li>The Reorderer, a JavaScript library for sorting DOM elements </li>
```
Please remove the space at the end of this line. The same issue appears several times in this file, please fix those as well

sachin10101998 replied:
These spaces were given so as to pass the markdown linter. Markdown gives linter error if one line has more than 120 characters in it. Therefore I broke the lines having more than 120 characters into two parts.

BlueSlug commented on src/posts/2009-03-02-Fluid-uPortal_Collaboration_2_0.md:
```diff
@@ -5,72 +5,64 @@ permalink: 2009-03-02-Fluid-uPortal_Collaboration_2_0.html
+          guidelines</li>
+         <li>Integrating Fluid&#39;s TransformAble services into the portal, providing a system for adapting the look
+          and feel of
+            uPortal based on user preferences
```
This line and the one above it are awkwardly separated. Is there a reason for this?

sachin10101998 replied:
Same as above. These spaces were given so as to pass the markdown linter. Markdown gives linter error if one line has more than 120 characters in it. Therefore I broke the lines having more than 120 characters into two parts. although I can improve the indentation at the beginning. however, the problem still lies when links are there. Some links go more than 120 characters,. and they break when I divide them

BlueSlug commented on src/posts/2009-05-27-Fluid_Infusion_1_0_released.md:
```diff
@@ -5,56 +5,75 @@ permalink: 2009-05-27-Fluid_Infusion_1_0_released.html
+         technologies, are fully controllable with the keyboard, and can be transformed to suit your users’
+         personal needs.
+         <br>
+         The Fluid Infusion 1.1 release includes a collection of our UI components, tutorials to help you get started,
```
This sentence probably isn't supposed to be in this post. I see that it's there on the production site. @jhung do you agree this line and line 76 should be removed?

sachin10101998 replied:
Leaving it for now.

BlueSlug commented on src/posts/2009-06-02-Fluid_Infusion_1_1_released.md:
```diff
@@ -5,57 +5,76 @@ permalink: 2009-06-02-Fluid_Infusion_1_1_released.html
+      <p>
+      <h3>What is Fluid Infusion?</h3>
+      </p>
+      <p>
```
There is a duplicate `<p>` opening on this line, please remove it

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2009-06-02-Fluid_Infusion_1_1_released.md:
```diff
@@ -5,57 +5,76 @@ permalink: 2009-06-02-Fluid_Infusion_1_1_released.html
+         <li>Changes some Framework API</li>
+         <li>Fixes many bugs</li>
+      </ul>
+      <p>
```
`h3` elements shouldn't be wrapped in `p` elements, please remove the `p` tags

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2009-06-02-Fluid_Infusion_1_1_released.md:
```diff
@@ -5,57 +5,76 @@ permalink: 2009-06-02-Fluid_Infusion_1_1_released.html
+      <p><strong> <a href="https://github.com/fluid-project/infusion">Download Fluid Infusion</a> </strong></p>
+      <h3>What&#39;s new in this release?</h3>
+      <p>
+      <p>This release:</p>
```
Why are there two sets of `p` tags?

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -5,53 +5,54 @@ permalink: 2009-08-12-Fluid_Infusion_1_1_1_released.html
+         designed to put you back in control of your application’s user experience. It includes a growing collection
+          of UI components
+         —reusable interactions that go deeper than most widgets. Created by a community of developers and interaction
+          designers,
```
The line breaks throughout this file are very awkward and make the markup difficult to follow

sachin10101998 replied:
Can improve the indentation once the markdown line-length issue is discussed.

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -5,53 +5,54 @@ permalink: 2009-08-12-Fluid_Infusion_1_1_1_released.html
+         Infusion components are built from the ground up with accessibility in mind. All of our designs can be used with
+         assistive technologies, are fully controllable with the keyboard, and can be transformed to suit your users’
+          personal
+         needs.
```
Same thing here. Why is this formatted this way?

sachin10101998 replied:
Line length issue again. Will resolve it after discussion.

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -5,53 +5,54 @@ permalink: 2009-08-12-Fluid_Infusion_1_1_1_released.html
+            the text edit.</a>
+         </li>
+         <li>
+            <a href="/browse/FLUID-1320">Typing &quot;Click here to edit&quot; is equivalent to erasing all the text</a>
```
This link should point to the Jira website like the one above it. The old link was wrong, too, but please fix it either way.

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -5,53 +5,54 @@ permalink: 2009-08-12-Fluid_Infusion_1_1_1_released.html
+         <li>
+            <a href="/browse/FLUID-1320">Typing &quot;Click here to edit&quot; is equivalent to erasing all the text</a>
+         </li>
+         <li> <a href="/browse/FLUID-2967">Item will drop to the right of an element when it should have dropped at the left.
```
This link should point to the Jira website.

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2009-08-12-Fluid_Infusion_1_1_1_released.md:
```diff
@@ -5,53 +5,54 @@ permalink: 2009-08-12-Fluid_Infusion_1_1_1_released.html
+         <li> <a href="/browse/FLUID-2967">Item will drop to the right of an element when it should have dropped at the left.
+            </a>
+         </li>
+         <li><a href="/browse/FLUID-3054">tiny_mce Inline Edit text editor fails when used with advanced theme</a></li>
```
This link should point to the Jira website.

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2010-09-08-Fluid_Infusion_1_2_1_and_1_1_3_releases.md:
```diff
@@ -5,58 +5,56 @@ permalink: 2010-09-08-Fluid_Infusion_1_2_1_and_1_1_3_releases.html
+          be used
+         with assistive technologies, are fully controllable with the keyboard, and can be transformed to suit your users’
+          personal
+         needs.
```
Please adjust the line breaks in this file

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2017-02-02-Infusion_2_0_released.md:
```diff
@@ -5,83 +5,102 @@ permalink: 2017-02-02-Infusion_2_0_released.html
+      <p>Infusion 2.0 includes significant framework improvements and <em>is not backwards compatible
+         </em> with previous versions of Infusion. Please see <a href="http://docs.fluidproject.org/infusion/
+            development/APIChangesFrom1_5To2_0.html">API Changes from 1.5 to 2.0</a> and <a href="http://docs.
+            fluidproject.org/infusion/development/DeprecationsIn1_5.html">Deprecations in 1.5</a> on the <a
```
This link is not working

sachin10101998 replied:
Fixed temporarily. But markdown linter breaks after this.

BlueSlug replied:
It seems to be passing the linter for me now. Is it still failing for you?

sachin10101998 replied:
Mo. It works for me too after latest commit.

BlueSlug commented on src/posts/2017-02-02-Infusion_2_0_released.md:
```diff
@@ -5,83 +5,102 @@ permalink: 2017-02-02-Infusion_2_0_released.html
+      <p>The Fluid community is pleased to announce the release of Infusion 2.0!</p>
+      <p>Infusion 2.0 includes significant framework improvements and <em>is not backwards compatible
+         </em> with previous versions of Infusion. Please see <a href="http://docs.fluidproject.org/infusion/
+            development/APIChangesFrom1_5To2_0.html">API Changes from 1.5 to 2.0</a> and <a href="http://docs.
```
This link is not working

sachin10101998 replied:
Fixed temporarily. But markdown linter breaks after this.

sachin10101998 commented on src/posts/2007-08-22-Off_to_a_Running_Start_with_UX_Walkthroughs.md:
```diff
@@ -0,0 +1,26 @@
+---
```
Fixed

sachin10101998 commented on src/posts/2007-05-04-Better_Web_Experiences.md:
```diff
@@ -5,90 +5,73 @@ permalink: 2007-05-04-Better_Web_Experiences.html
+      </p>
+   </div>
+   <div class="medium-6 columns">
+      <p>The <strong>Fluid Project</strong> (<a href="http://fluidproject.org)">http://fluidproject.org)</a>,
```
Fixed

BlueSlug replied:
This still seems to be broken. Have you synced with remote?

sachin10101998 commented on src/posts/2007-05-04-Better_Web_Experiences.md:
```diff
@@ -5,90 +5,73 @@ permalink: 2007-05-04-Better_Web_Experiences.html
+      </p>
+      <p>Many of the Fluid Project partners are founding contributors to the academic community source software
+         movement, and bring a wealth of design and development expertise to the project. Please visit
+         [<a href="http://fluidproject.org]">http://fluidproject.org]</a> for more information or to get involved.
```
Fixed

sachin10101998 commented:
@BlueSlug  I found a workaround to deal with styling while keeping all the posts in markdown. I have made the required changes in both floe and fluid. Please check.

BlueSlug commented on src/posts/2009-05-27-Fluid_Infusion_1_0_released.md:
```diff
@@ -0,0 +1,50 @@
+- Provides the ability to create custom builds
+- Adds jQuery UI Themes for working with FSS themes
+- Adds new and powerful decorators for the Renderer:
+      - The &quot;fluid&quot; decorator instantiates any Fluid Infusion component bound to the markup
```
The `tab` characters in this nested list and the ones below are causing the formatting to become strange:
![image](https://user-images.githubusercontent.com/547933/89695127-e5544300-d8e0-11ea-83a8-6b7990b96c3d.png)

It should look like this:
![image](https://user-images.githubusercontent.com/547933/89695134-ec7b5100-d8e0-11ea-97af-c075b1686e49.png)

sachin10101998 replied:
fixed

BlueSlug commented on src/posts/2008-08-01-Fluid_Infusion_v0_4_released.md:
```diff
@@ -0,0 +1,28 @@
+New features of this release include:
+
+- Preview versions of two new components:</br>
+      - Inline Edit</br>
```
The `tab` characters in this nested list are causing the formatting to become strange. Please remove them and the `</br>` tags, and the list should look correct after that

sachin10101998 replied:
Fixed

BlueSlug commented on src/posts/2008-06-02-Fluid_Infusion_v0_3_released.md:
```diff
@@ -0,0 +1,28 @@
+tags: post
+author: Release Manager
+---
+**The Fluid Project is pleased to announce the release of Fluid Infusionv0.3, the latest release of the**
```
Please remove the `**` at the end of this line

sachin10101998 replied:
fixed

BlueSlug commented on src/posts/2008-06-02-Fluid_Infusion_v0_3_released.md:
```diff
@@ -0,0 +1,28 @@
+author: Release Manager
+---
+**The Fluid Project is pleased to announce the release of Fluid Infusionv0.3, the latest release of the**
+**Fluid  component library and User Experience (UX) Toolkit.
```
Please remove the `**` at the beginning of this line and add it at the end, the whole paragraph is meant to be bold.

sachin10101998 replied:
fixes

BlueSlug commented on src/posts/2007-12-01-Fluid 0_1_is_Here.md:
```diff
@@ -0,0 +1,37 @@
+
+### What&#39;s New in This Release?</h3>
+
+- The Reorderer, a JavaScript library for sorting DOM elements ###
```
Please remove the `###` from the end of this line and the four lines after it

sachin10101998 replied:
fixed

BlueSlug commented on src/posts/2007-12-01-Fluid 0_1_is_Here.md:
```diff
@@ -0,0 +1,37 @@
+Fluid is still in early development, and this release has a number of  known issues. API changes will be coming
+in the future, so for now, we  encourage you to use 0.1 for experimentation and prototyping. Check the
+README.txt file in the download package for a list of known issues.
+Thanks to everyone in the community for their hard work on this release!
```
Please add an extra line break before this line, this is meant to be two paragraphs

sachin10101998 replied:
fixed

BlueSlug commented on src/posts/2007-12-01-Fluid 0_1_is_Here.md:
```diff
@@ -0,0 +1,37 @@
+The Fluid Project is pleased to announce the first release of the Fluid component library and
+User Experience (UX) Toolkit.
+
+### What&#39;s New in This Release?</h3>
```
Please remove all instances of `</h3>` from this file. Search-and-replace can often lead to leftover bits of markup like this, it's important to check afterwards :)

sachin10101998 replied:
fixed

BlueSlug commented on Gruntfile.js:
```diff
@@ -14,90 +16,117 @@ module.exports = function (grunt) {
+            sources: {
+                json: ["package.json", ".eslintrc.json","./src/_data/*.json"],
+                js: ["./src/assets/js/*.js","./src/transforms/*.js","./src/filters/*.js","./src/assets/js/*.js","./src/filters/*.js","./src/utils/*.js",".eleventy.js","Gruntfile.js"],
+                md: ["./src/posts/*.md"]
```
Please include all other Markdown files in the project in the md linter (README.md, AUTHORS.md, CONTRIBUTING.md, etc.), except for the news post template and anything in the `lib` directory

sachin10101998 replied:
Added README.md, AUTHORS.md, CONTRIBUTING.md, etc.. The lib folder is not added because it's being copied from infusion and we lint only third party files. We discussed this earlier and came to a conclusion that this folder does not need to be linted. Therefore, I didn't add it.

BlueSlug replied:
Great, thanks!

Other files that I would suggest adding to this list, other than the three you just mentioned, would be:
- `./_data/README.md`
- `./.github/ISSUE_TEMPLATE/bug_report.md`
- `./.github/ISSUE_TEMPLATE/feature_request.md`

and *maybe* `./src/assets/stylesheets/foundation/README.md` since it's not copied in as a 3rd-party file, but I'm less concerned about that one.

sachin10101998 commented on src/_includes/layouts/default.html:
```diff
@@ -0,0 +1,31 @@
+<!doctype html>
+<html lang="en" dir="ltr">
+    <head>
+        <title>{% if title %} {{ title }} {% else %} {{ site.title }} {% endif %}</title>
```
Fixed

sachin10101998 commented on src/_includes/layouts/newsindex.html:
```diff
@@ -0,0 +1,31 @@
+<!doctype html>
+<html lang="en" dir="ltr">
+    <head>
+        <title>{% if title %} {{ title }} {% else %} {{ site.title }} {% endif %}</title>
```
Fixed

sachin10101998 commented on src/_includes/layouts/post.html:
```diff
@@ -0,0 +1,41 @@
+<!doctype html>
+<html lang="en" dir="ltr">
+    <head>
+        <title>{% if title %} {{ title }} {% else %} {{ site.title }} {% endif %}</title>
```
Fixed

sachin10101998 commented on src/index.html:
```diff
@@ -0,0 +1,45 @@
+---
+layout: layouts/default
+title: The Open Source Software Community - Fluid Project
```
Fixed

sachin10101998 commented on src/infusion.html:
```diff
@@ -0,0 +1,73 @@
+---
+layout: layouts/default
+permalink: infusion.html
+title: Infusion | fluid
```
Fixed

sachin10101998 commented on src/about.html:
```diff
@@ -0,0 +1,41 @@
+---
+layout: layouts/default
+title: About | fluid
```
Fixed

sachin10101998 commented on src/posts/index.html:
```diff
@@ -0,0 +1,19 @@
+---
+layout: layouts/newsindex
+title: News
```
Fixed

sachin10101998 commented on src/projects.html:
```diff
@@ -0,0 +1,127 @@
+---
+layout: layouts/default
+title: Projects | fluid
```
Fixed

sachin10101998 commented on src/posts/2017-02-02-Infusion_2_0_released.md:
```diff
@@ -0,0 +1,65 @@
+author: Release Manager
+---
+The Fluid community is pleased to announce the release of Infusion 2.0!
+Infusion 2.0 includes significant framework improvements and **is not backwards compatible**
```
fixed

sachin10101998 commented on src/posts/2013-03-05-Fluid_Releases_New_Video_Player.md:
```diff
@@ -0,0 +1,27 @@
+author: Colin Clark
+---
+The Fluid community is happy to announce the release of Video Player v0.1.
+Video Player is designed to be a fully accessible HTML5 video player.
```
fixed

sachin10101998 commented on src/posts/2010-12-23-Infusion_1_3_is_here.md:
```diff
@@ -0,0 +1,61 @@
+
+- Sneak Peek at the new Inversion of Control (IoC) system
+- Uploader support for HTML 5
+      - Flash-free for modern browsers!
```
fixed

sachin10101998 commented on src/posts/2009-06-02-Fluid_Infusion_1_1_released.md:
```diff
@@ -0,0 +1,51 @@
+- Provides the ability to create custom builds
+- Adds jQuery UI Themes for working with FSS themes
+- Adds new and powerful decorators for the Renderer:
+      - The &quot;fluid&quot; decorator instantiates any Fluid Infusion component bound to the markup
```
fixed

sachin10101998 commented:
@BlueSlug I have made the required changes.

gtirloni commented:
> @gtirloni: is there anything that should be added for deployment to work properly once this is merged in? There are already the files you added to build the image and push it to Docker Hub, of course.

I'll need to configure the servers to properly deploy this and the changes aren't very simple. I'd prefer to work on this in a separate PR, if that's okay.

BlueSlug commented:
@gtirloni: I have no objection to that being covered in a separate PR as long as merging this into `fluid-project/master` won't impact the deployed site :)

gtirloni commented:
@BlueSlug excellent point! I've disabled the existing Jenkins jobs for the fluid and floe websites so it won't deploy anything. Thanks!

jhung commented on package.json:
```diff
@@ -1,35 +1,43 @@
+        "lint": "grunt lint",
+        "clean": "rimraf dist",
+        "cleandependencies": "rimraf node_modules",
+        "cleanall": "rimraf node_modules && rimraf dist"
```
I can not seem to successfully run `cleanall`. To reproduce:

1. Run `npm i` then `npm run build`
2. Run `npm run cleanall`
 The following would appear in the error log:
```
13 verbose stack Error: fluidproject.org@1.0.0 cleanall: `rimraf node_modules && rimraf dist`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (C:\Users\jhung\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:223:5)
13 verbose stack     at ChildProcess.<anonymous> (C:\Users\jhung\AppData\Roaming\npm\node_modules\npm\node_modules\npm-lifecycle\lib\spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:223:5)
13 verbose stack     at maybeClose (internal/child_process.js:1021:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:283:5)
```

sachin10101998 replied:
I recreated the scenario and I do not see this error.

jhung replied:
Thanks for testing. I've filed an issue (#61 ) so it can be investigated later.

sachin10101998 commented on README.md:
```diff
@@ -1,20 +1,21 @@
 1. Get the required node modules: `npm install`
-2. Run docpad from the fluid-website directory `npm run docpad`.
-3. Open `http://localhost:9778/` to see the website. 
+2. Run [11ty](http://11ty.dev) from the fluid-website directory `npm run eleventyport`.
```
fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,84 @@
+      `layout: layouts/post`
+      `title: Title of News Article`
+      `date: 'YYYY-MM-DD'`
+      `tags: post`
```
Fixed

sachin10101998 commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,84 @@
+      `title: Title of News Article`
+      `date: 'YYYY-MM-DD'`
+      `tags: post`
+      `permalink: YYYY-MM-DD-newsArticleName.html`
+      `filename: Short URL for news. May contain Capital Items but no spaces`
```
fixed

jhung commented:
@gtirloni This PR is now ready for deploy testing. Let me know what the next steps are.

gtirloni commented:
> @gtirloni This PR is now ready for deploy testing. Let me know what the next steps are.

Thanks! I'll start working on that tomorrow morning.
